### ISSUE TITLE
Accessibility Design Review Feedback for Caregiver 10-10CG MVP

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#8576]
## Caregiver Design Review - Accessibility Findings

**Feedback framework**
- **❗️ Must** for if the feedback must be applied 
- **⚠️Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Point of Contact

**VFS Point of Contact:** Jennifer

@1Copenut — ready for your review

## Definition of Done
- PM and designer acknowledge feedback
- Document decisions made
- VSA accessibility specialist will review and close ticket

<hr>

### Throughout

* ❗ **Must:** The Need help? component used on these pages differs from the one used elsewhere. [documented in issue tickets #8556, #8558)]

<hr>

### Step 1 Veteran/service member information

1. **Should:** Veteran's sex (*Required) — **508 issue: Cognition [WCAG 2.0 - 3.2.4 Consistent Identification](https://www.w3.org/TR/UNDERSTANDING-WCAG20/consistent-behavior-consistent-functionality.html)**
    Forms throughout VA.gov paper forms were updated to use the Gender dropdown. This meets guidelines from GSA to provide a standardized experience across federal services. Providing consistency across VA.gov is vital to the well-being of our Veterans who have a greater incidence of cognitive considerations leading to reading comprehension difficulties, emotional overwhelm for what may be trivial to others. 

    Providing the three options shown in the prototype is also encouraged, as a number of Veterans may change their gender identity after discharge, and invalidation may trigger cognitive considerations symptoms. 

    #### 10-10EZ, for reference

    ![caregiver-sex-01](https://user-images.githubusercontent.com/57469/80632011-9e72c980-8a24-11ea-98fd-bff820c77e60.png)

    ![caregiver-sex-02](https://user-images.githubusercontent.com/57469/80632033-a468aa80-8a24-11ea-8275-eb1249b5efc9.png)

1. **Should:** VA medical center information — **accessibility issue: Cognition – interaction design best practice; [WCAG 2.1 - 1.4.10 Reflow](https://www.w3.org/TR/WCAG21/#reflow)**
    The text input for the name of the medical facility should be wider to accommodate the length of facility names without cut-off. This may also be an issue for the VA Medical Center dropdown further down the page. Name of the facility should not be cut off in the selected state. The text inputs for the Veteran and Caregiver information are all longer than those provided for VAMC names. Just as the size of a Zip Code input is smaller than that of a street address, it is standard practice to size text inputs sufficiently to allow someone to see the content they're inputting without cut-off, when possible.

    #### Screenshot of input, for reference

    ![caregiver-vamc-input-length](https://user-images.githubusercontent.com/57469/80632441-4be5dd00-8a25-11ea-949a-c73980be46f1.png)

1. **Should:**  "Can't find your medical center or clinic? Show/hide —  **accessibility issue: Mobility, Cognition, Screen reader – Interaction design best practice**
    The information contained in this show/hide is short enough, and useful to those filling out the form (which is unlikely to be a repeated activity) that it should be displayed on the page without a click to reveal.

    Before hiding content, be certain of why you want to hide it. Is the amount of content enough that hiding it helps the user? If it is a page where users come to once, avoid hiding the info. If they come to the page repeatedly, there may be more of a case for hiding it. To reveal the content naturally takes some motor and cognitive effort on the part of users. Be certain that their effort is rewarded and valued.

<hr>

### Step 2 Primary Family Caregiver information

1. ❗ **Must:**  What's a Primary Family Caregiver additional info — **508 issues:  Screen reader, Cognition – [WCAG 2.0 - 3.1.3 Unusual Words](https://www.w3.org/TR/UNDERSTANDING-WCAG20/meaning-idioms.html), [Plain Writing Act of 2010](https://www.govinfo.gov/app/details/PLAW-111publ274)**
    The legal language here does not meet the criteria spelled out in the [Plain Writing Act of 2010](https://www.govinfo.gov/app/details/PLAW-111publ274), which requires Government documents issued to the public must be written clearly. The current [paper 10-10CG](https://www.va.gov/health-care/forms/vha-10-10CG.pdf) language does not meet the criteria. The [plain language]() information contained in the modal should be presented in the show/hide, and if necessary the legal language provided at the bottom of that show/hide. The preference would be not to hide this information in a show/hide component, as it will be useful to anyone completing this form.
    From the WCAG success criteria: Certain disabilities make it difficult to understand nonliteral word usage and specialized words or usage. Certain disabilities make it difficult to understand figurative language or specialized usage. Providing such mechanisms is vital for these audiences. Specialized information intended for non-specialist readers is encouraged to satisfy this Success Criterion, even when claiming only Single-A or Double-A conformance.

1. **Should:** Modal inside of show/hide — **accessibility issues:  Screen reader, Cognition, Mobility – Interaction design best practice, will incur additional QA risk**
    Putting a modal inside of a show/hide is a poor usability / accessibility pattern. For our Veterans with cognition and mobility impairments this level of fine motor control will be challenging. For technical implementation, focus management will need to be custom-built and QA'd, but due to the custom nature will be a risk for edge case bugs.

1. ❗ **Must:** Proximity of links — **accessibility issues:  Mobility – Interaction design best practice**
    The Learn more about what this means link is close to the legal statute link and may introduce errors. Interaction design best practice would recommend 1.5rem space between, in this case 24 pixels.

1. ❗ **Must:**  Link purpose issues  — **508 issues:  Screen reader, Cognition, Mobility – [WCAG 2.0 2.4.9 LInk Purpose, link only](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-link.html), [WCAG 2.0 2.4.4 Link Purpose, in context](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-refs.html)**
    The label "Learn more about what this means" should read "Learn more about what a Primary Family Caregiver Is" in order to meet 508 Success Criteria, "The purpose of each link MUST be determined from the link text alone." In this instance it will be confusing for those with cognitive considerations to essentially duplicate the link that opened the show/hide. 

1. ❗ **Must:**  Placeholder text? — **508 issue: Cognition [WCAG 2.0 - 3.2.4 Consistent Identification](https://www.w3.org/TR/UNDERSTANDING-WCAG20/consistent-behavior-consistent-functionality.html)**
    Under Contact information, there is placeholder text in the phone number fields which must be removed to align with the [VA Design System](https://design.va.gov/components/form-controls#accessibility). WCAG 2.0 requires that "components that have the same functionality within a set of Web pages are identified consistently."

1. **Should:** Health care coverage — **508 issues:  cognition, mobility [WCAG 2.0 - 3.2.4 Consistent Identification](https://www.w3.org/TR/UNDERSTANDING-WCAG20/consistent-behavior-consistent-functionality.html), [Reducing cognitive overload, article](https://www.smashingmagazine.com/2016/09/reducing-cognitive-overload-for-a-better-user-experience/), [WCAG 2.1 - 2.5.1 Complex Pointer Gestures](https://www.w3.org/WAI/standards-guidelines/wcag/new-in-21/#251-pointer-gestures-a)**
    Under Health care coverage, provide the health care programs without the show/hide to support the cognition needs of those completing the form. The cognitive load to remember the definitions as they continue to work through the form may be a burden to our users with TBI, PTSD, memory, or other cognitive considerations. 
    - The design with the lengthy and similar labels provides a comprehension challenge for those with cognitive considerations. 
    - The number of radio inputs to discern and then manipulate adds burden for those with mobility issues. 
    - For mobile users this interaction pattern will prove especially difficult to understand the individual labels, when the user is likely to see one at a time on the screen. For those with cognitive considerations on mobile devices, it will be hard to remember.

![Screen Shot 2020-04-29 at 6 22 14 PM](https://user-images.githubusercontent.com/57469/80652673-68ded800-8a46-11ea-9baf-87b3ac657e8e.png)

  >    Is the Primary Family Caregiver currently enrolled in Medicaid?
  >    Is the Primary Family Caregiver currently enrolled in Medicare?
  >    Is the Primary Family Caregiver currently enrolled in Tricare?
  >    Is the Primary Family Caregiver currently enrolled in CHAMPVA?
  >    Is the Primary Family Caregiver currently enrolled in other health insurance?

  A solution that may be more helpful for those with disabilities or on mobile devices :

Please select the health care coverage Primary Family Caregiver currently has.

- [ ] Medicaid 
       (a government health program for eligible low-income individuals and families and people with disabilities)
- [ ] Medicare 
       (a federal health insurance program providing coverage for people who are 65 years or older or who meet special criteria. Part A insurance covers hospital care, skilled nursing and nursing home care, hospice, and home health services)
- [ ] Tricare 
       (a government health care program for service members, veterans, and their families)
- [ ] CHAMPVA 
       (a cost-sharing program that covers the price of some health care services and supplies)
- [ ] Other health insurance
- [ ] No health insurance coverage

  The definitions could be provided in a tooltip, that on mobile would be absolutely positioned 24px from the left edge of the viewport, with a fixed width. 
  If the Yes and No responses are required to be called out, it would be helpful to revisit the labels, for example:

    Please select the health care coverage Primary Family Caregiver currently has.

    Medicare
    Yes
    No
    Medicaid
    Yes
    No
    Tricare
    Yes
    No
    CHAMPVA
    Yes
    No 
    Other health insurance
    Yes
    No  

    This is still likely to be a burden on those with cognitive considerations and those on mobile devices.

<hr>

### Step 3 - Secondary Family Caregiver information

1. Same issues as above with the "What's a Secondary Family Caregiver?" show/hide and modal, as indicated in the Primary Family Caregiver information.

1. Same issues as above with the Secondary Family Caregiver's sex question.

1. **Should:**  Secondary Family Caregiver (2) — **508 issue: cognition [WCAG 2.0 - 3.2.4 Consistent Identification](https://www.w3.org/TR/UNDERSTANDING-WCAG20/consistent-behavior-consistent-functionality.html)**
    For the Secondary Family Caregiver (2) information, each of the information fields should include the (2) to provide consistent identification for cognitive considerations, screen readers, and those who may be using translation tools to read the form.

<hr>

### Step 4 — Review application and signature

1. ❗ **Must:**  Statement of truth text input labels — 508 issue: screen reader, cognition [WCAG 2.0 - 2.4.4 Link Purpose, in context](https://www.w3.org/WAI/WCAG21/Understanding/link-purpose-in-context.html)
    For the text input currently labeled, "Veteran or service member's full name", should be "Enter Veteran or service member's full name". The screen reader will not indicate what is expected in this input field. Same issue with both Caregiver statement of truth text inputs. In addition to be text that explains an input, input labels are links that communicate a purpose to select the input.

**Comment by jenstrickland** [comment]
After I documented the items above, I wondered about error states for the healthcare coverage and tested staging. The error state for the string of radio inputs could be even more confusing than the unselected state. 

What's in staging only shows an error on the last question, but it should display an error on all of the inputs, which would be tough for anyone to figure out. 
![Screen Shot 2020-04-29 at 8 16 31 PM](https://user-images.githubusercontent.com/57469/80659262-7439ff80-8a56-11ea-83ce-fbea0707e5c9.png)

**Comment by 1Copenut** [comment]
I agree with the items @jenstrickland has raised. VSP has nothing to add.
